worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/segment/SegmentMapPartitionDataReader.java

```diff
@@ -276,9 +276,10 @@ private void updateSegmentId(int subPartitionId) {
       }
       if (segmentId != -1) {
         // For the continuous segments, we use the same backlog
+        Integer subPartitionLastSegmentId = subPartitionLastSegmentIds.get(subPartitionId);
```

Comment:
Will (Can ?) this be null only when segmentId == 0 ?
Else the `!segmentId.equals(subPartitionLastSegmentId + 1)` will throw NPE

Reply:
@mridulm, line 279 does not return null, which could not throw NPE. You could refer to https://github.com/apache/celeborn/pull/2820#discussion_r1818441685. cc @reswqa.

Reply:
In that case, cant we not simply use `int` instead of `Integer` ?

Reply:
@mridulm, yes. I updated this. PTAL.